Source/NSubstitute/Core/CallSpecificationFactory.cs

```diff
@@ -15,7 +15,7 @@ public ICallSpecification CreateFrom(ICall call, MatchArgs matchArgs)
         {
             var methodInfo = call.GetMethodInfo();
             var argumentSpecs = call.GetArgumentSpecifications();
-            var arguments = call.GetArguments();
+            var arguments = call.GetOriginalArguments();
```

Comment:
That's the necessary change, unfortunately. Otherwise, no way for user to configure results using the `Returns()` extension if my handler fills the `ref` arguments.
If you see any issues with that - let me know. So far no tests failed :)